src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -147,5 +154,106 @@ private static TypeKind GetNewTypeKind(ImmutableArray<ITypeParameterSymbol> type
 
             return docEditor.GetChangedDocument();
         }
+
+        private static async Task<Solution> RefactorReferencesAsync(
+            ILookup<DocumentId, (ReferenceLocation location, bool isExtension)> locationsToDoc,
+            Solution solution,
+            INamedTypeSymbol newType,
+            CancellationToken cancellationToken)
+        {
+            var solutionEditor = new SolutionEditor(solution);
+            foreach (var docGroup in locationsToDoc)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var docEditor = await solutionEditor.GetDocumentEditorAsync(docGroup.Key, cancellationToken).ConfigureAwait(false);
+                FixReferencesSingleDocument(docGroup.AsImmutable(), docEditor, newType, cancellationToken);
+            }
+
+            return solutionEditor.GetChangedSolution();
+        }
+
+        private static void FixReferencesSingleDocument(
+            ImmutableArray<(ReferenceLocation location, bool isExtension)> referenceLocations,
+            DocumentEditor docEditor,
+            INamedTypeSymbol newType,
+            CancellationToken cancellationToken)
+        {
+            foreach (var (refLoc, isExtension) in referenceLocations)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // Candidate locations represent a non-perfect match (like incorrect number of params),
+                // which means it likely isn't the member access we're looking for. It's either a reference to
+                // something else, like an overload, in which case we shouldn't change it, or it's an error
+                // by the user, meaning it's ok if we don't change it, as the user will have to fix it anyways
+                if (refLoc.IsCandidateLocation)
+                {
+                    continue;
+                }
+
+                var doc = docEditor.OriginalDocument;
+
+                var syntaxFacts = doc.GetRequiredLanguageService<ISyntaxFactsService>();
+                var refNode = docEditor.OriginalRoot.FindNode(refLoc.Location.SourceSpan, findInsideTrivia: true, getInnermostNodeForTie: true);
+                // track this node in case the doc has changed
+                docEditor.ReplaceNode(refNode, refNode.TrackNodes(refNode));
+
+                // add imports for each reference as there may be multiple locations that we need to add to
```

Comment:
this feels like this could really break code as it will pull in symbols that may cause conflicts at that location.